flagconf.go

```diff
@@ -54,7 +56,25 @@ func ParseStrings(args []string, path string, config interface{}, allowNoConfigF
 	}
 
 	// Override any settings with configured flags
-	return flagset.Parse(args[1:])
+	if err = flagset.Parse(args[1:]); err != nil {
+		// In case flag parsing fails, return a custom error containing usage
+		// info if user wants to print it.
+		buf := &bytes.Buffer{}
+		flagset.SetOutput(buf)
```

Comment:
Because this is after `flagset.Parse`, it won't capture the printing of the error. For instance, this code:

``` go
if err := flagconf.Parse("config.toml", &config); err != nil {
	fmt.Println(err)
}
```

makes this happen:

```
$ ./prog  -asdf
flag provided but not defined: -asdf
flag provided but not defined: -asdf
```

Is that intentional?